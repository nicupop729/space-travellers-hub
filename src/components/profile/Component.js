import PropTypes from 'prop-types';
import MissionBtn from '../Btns/MissionBtn';
import RocketBtn from '../Btns/RocketBtn';
import './component.css';

const Component = ({
  reservations, object, message, type,
}) => (
  <div className="Reservations__div">
    <h1>
      My
      {object}
    </h1>
    {reservations.length ? (
      <ul className="Reservations__div_list">
        {reservations.map(({
          name, id, urlWiki, urlOwn,
        }) => (
          <li className="Reservations__div_list_item" key={id}>
            <h2 className="Reservations__div_list_item_name">
              {name}
              {' '}
              -
              {' '}
              {urlWiki && <a href={urlWiki} target="_blank" rel="noreferrer">Wiki</a>}
              {' '}
              -
              {' '}
              {urlOwn && <a href={urlOwn} target="_blank" rel="noreferrer">Own Site</a>}
              -
              {' '}
            </h2>
            {type === 'mission' ? (
              <MissionBtn
                className="app-btn-danger"
                text="Leave Mission"
                id={id}
              />
            ) : (
              <RocketBtn
                className="Rocket__btn_Cancel"
                text="Cancel Reservation"
                id={id}
              />
            )}
          </li>
        ))}
      </ul>
    ) : (
      <p className="Reservations__div_none">{message}</p>
    )}
  </div>
);

Component.propTypes = {
  reservations: PropTypes.instanceOf(Object).isRequired,
  object: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
};

export default Component;
