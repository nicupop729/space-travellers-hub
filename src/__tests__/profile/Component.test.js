import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import Component from '../../components/profile/Component';

describe('snapshot testing', () => {
  test('snapshot for Component component', () => {
    const activeRockets = [
      { test1: 1, test2: 2 },
      { test3: 3, test4: 4 },
    ];
    const rendererComponent = renderer
      .create(
        <Provider store={store}>
          <Component reservations={activeRockets} />
        </Provider>,
      )
      .toJSON();
    expect(rendererComponent).toMatchSnapshot();
  });
});
