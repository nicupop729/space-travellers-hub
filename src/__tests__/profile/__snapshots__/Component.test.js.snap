// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing snapshot for Component component 1`] = `
<div
  className="Reservations__div"
>
  <h1>
    My
  </h1>
  <ul
    className="Reservations__div_list"
  >
    <li
      className="Reservations__div_list_item"
    >
      <h2
        className="Reservations__div_list_item_name"
      >
         
        -
         
         
        -
         
        -
         
      </h2>
      <button
        className="Rocket__btn_Cancel"
        onClick={[Function]}
        type="button"
      >
        Cancel Reservation
      </button>
    </li>
    <li
      className="Reservations__div_list_item"
    >
      <h2
        className="Reservations__div_list_item_name"
      >
         
        -
         
         
        -
         
        -
         
      </h2>
      <button
        className="Rocket__btn_Cancel"
        onClick={[Function]}
        type="button"
      >
        Cancel Reservation
      </button>
    </li>
  </ul>
</div>
`;
